back/engines/commercial/insights/app/services/insights/xlsx_service.rb

```diff
@@ -0,0 +1,30 @@
+module Insights
+	class XlsxService
+
+		def generate_inputs_xlsx(inputs, categories, view_private_attributes = false)
+      columns = xlsx_service.generate_idea_xlsx_columns(inputs, view_private_attributes: view_private_attributes)
+
+
+
+      categorized = inputs.includes(:insights_category_assignments)
+
+      categories.each { |category|
+        columns.insert(3, { header: category.name, f: -> (i) {
+            icas = i.insights_category_assignments.where(category_id: category.id)
+            return nil if icas.blank?
+            icas.first.approved ? 'approved' : 'suggested'
+        }})
+      }
+
+      xlsx_service.generate_xlsx 'Inputs export', columns, categorized
+    end
```

Comment:
(Just a suggestion -- mostly 💅 changes, the only difference I think is that it maintains the order of categories 

```suggestion
    def generate_inputs_xlsx(inputs, categories, view_private_attributes: false)
      columns = xlsx_service.generate_idea_xlsx_columns(inputs, view_private_attributes: view_private_attributes)

      category_cols = categories.map do |category|
        {
          header: category.name,
          f: lambda do |input|
            assignment = input.insights_category_assignments.find_by(category_id: category.id)
            return unless assignment

            assignment.approved ? 'approved' : 'suggested'
          end
        }
      end

      columns.insert(3, *category_cols)
      inputs = inputs.includes(:insights_category_assignments)
      xlsx_service.generate_xlsx('Inputs export', columns, inputs)
    end
```